airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -276,12 +276,8 @@ def post_set_task_instances_state(dag_id, session):
         raise BadRequest(detail=str(err.messages))
 
     error_message = f"Dag ID {dag_id} not found"
-    try:
-        dag = current_app.dag_bag.get_dag(dag_id)
-        if not dag:
-            raise NotFound(error_message)
-    except SerializedDagNotFound:
```

Comment:
`get_dag()` still can raise `SerializedDagNotFound` after this change, so this `except` still needs to exist.